lesson1/assignment/yours.sol

```diff
@@ -1 +1,77 @@
 /*作业请提交在这个目录下*/
+pragma solidity ^0.4.14;
+
+contract Payroll {
+    uint constant payDuration = 10 seconds;
+
+    address owner;
+    uint salary;
+    address employee;
+    uint lastPayday;
+
+    function Payroll() {
+        owner = msg.sender;
+    }
+
+    function updateEmployee(address e, uint s) {
+        require(msg.sender == owner);
+
+        if (employee != 0x0) {
+            uint payment = salary * (now - lastPayday) / payDuration;
+            employee.transfer(payment);
+        }
+
+        employee = e;
+        salary = s * 1 ether;
+        lastPayday = now;
+    }
+
+    function updateEmployeeOnly(address e) {
+        require(msg.sender == owner&&e != 0x0);
```

Comment:
The employer should not have the privilege to keep changing employees without clearing the remaining payment.

Reply:
Ok, added the clearPayment function